src/algorithms/calorimetry/CalorimeterHitDigi.cc

```diff
@@ -96,11 +98,25 @@ void CalorimeterHitDigi::init() {
     // all these are for signal sum at digitization level
     if (!m_cfg.fields.empty()) {
         for (auto & field : m_cfg.fields) {
-            id_inverse_mask |= id_desc.field(field)->mask();
+            id_inverse_mask |= id_spec.field(field)->mask();
         }
         debug("ID mask in {:s}: {:#064b}", m_cfg.readout, id_mask);
     }
     id_mask = ~id_inverse_mask;
+
+    std::function hit_to_map = [this](const edm4hep::SimCalorimeterHit &h) {
```

Comment:
Optional, deferrable. Can this be reused from CalorimeterHitReco, somehow? It's essentially the same code. Templated lambda with concept `has_cellID`?

Reply:
That is understood. What goes into the function is still in flux at this point, will likely add more fields or change how this works anyway.